src/Actions/Contexts/ElementContext.cs

```diff
@@ -71,7 +71,7 @@ public ElementDataContext DataContext
                 return _dataContext;
             }
 
-            set
+            internal set
```

Comment:
I'm a bit concerned that someone might incorrectly try to use this method in production code. Should we consider adding a simple wrapper (something like SetElementContextForUnitTests) that then calls the setter?

Reply:
I like it. Will do.

Reply:
Turns out this change wasn't necessary for the commit. I've reverted it.